downstream distro: No rule to make target 'nokogiri.so'

Hi!

Context: I am trying to create installable [ruby bootstraps](http://github.com/jekyllex/ruby-android) for [android architectures](https://developer.android.com/ndk/guides/abis). I have [altered](https://github.com/jekyllex/ruby-android/blob/main/patches/ruby/bundled_gems.patch) the default bundled gems so that I can customize what gems are pre-installed. I wish to include nokogiri as well. To do so I use the the [termux-packages](https://github.com/termux/termux-packages) build system to build the dependencies `zlib`, `libiconv`, `libxml2`, `libxslt`.

The build system places the built files at `/data/data/<app-package>/files/usr/`, which is auto included into the flags when I build [ruby](https://github.com/termux/termux-packages/tree/master/packages/ruby) from their patches. So enabling `--use-system-libraries` does pick up the dependencies during the build.

So far things look fine as the parts are being compiled correctly. But I get an error at the final step

```
...
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_cdata.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_comment.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_document.c
/home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_document.c:92:7: warning: '__xmlDeregisterNodeDefaultValue' is deprecated [-Wdeprecated-declarations]
  if (xmlDeregisterNodeDefaultValue) {
      ^
...
/data/data/sh.gourav.jekyllex/files/usr/include/libxml2/libxml/tree.h:682:10: note: expanded from macro 'XML_GLOBALS_TREE'
         XML_DEPRECATED)
         ^
/data/data/sh.gourav.jekyllex/files/usr/include/libxml2/libxml/xmlexports.h:69:43: note: expanded from macro 'XML_DEPRECATED'
    #define XML_DEPRECATED __attribute__((deprecated))
                                          ^
1 warning generated.
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_document_fragment.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_dtd.c
...
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_namespace.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_node.c
/home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_node.c:2176:3: warning: 'htmlHandleOmittedElem' is deprecated [-Wdeprecated-declarations]
  htmlHandleOmittedElem(0);
  ^
/data/data/sh.gourav.jekyllex/files/usr/include/libxml2/libxml/HTMLparser.h:171:1: note: 'htmlHandleOmittedElem' has been explicitly marked deprecated here
XML_DEPRECATED
^
/data/data/sh.gourav.jekyllex/files/usr/include/libxml2/libxml/xmlexports.h:69:43: note: expanded from macro 'XML_DEPRECATED'
    #define XML_DEPRECATED __attribute__((deprecated))
...
/data/data/sh.gourav.jekyllex/files/usr/include/libxml2/libxml/xmlexports.h:69:43: note: expanded from macro 'XML_DEPRECATED'
    #define XML_DEPRECATED __attribute__((deprecated))
                                          ^
2 warnings generated.
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_node_set.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_processing_instruction.c
...
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xml_xpath_context.c
compiling /home/builder/.termux-build/ruby/src/.bundle/gems/nokogiri-1.16.5/ext/nokogiri/xslt_stylesheet.c
linking shared-object nokogiri/nokogiri.so
ln -sfT ../../../../src/.bundle/gems/nokogiri-1.16.5/lib ../../../../../.bundle/gems/nokogiri-1.16.5/lib
make[2]: *** No rule to make target 'nokogiri.so', needed by 'clean-ports'.  Stop.
make[2]: Leaving directory '/home/builder/.termux-build/ruby/build/.bundle/gems/nokogiri-1.16.5/ext/nokogiri'
make[1]: *** [exts.mk:338: .bundle/gems/nokogiri-1.16.5/ext/nokogiri/all] Error 2
make[1]: Leaving directory '/home/builder/.termux-build/ruby/build'
make: *** [uncommon.mk:397: build-ext] Error 2
```

Could anyone please help point out what might be going on here? The build was performed in GitHub actions so I don't have access to `.bundle/gems/nokogiri-1.16.5/ext/nokogiri` folder where the error logs might be. Since getting the logs can be tricky (temporary folders are cleaned up the build system) and the build take a lot of time, I'll try to put them here as soon as I am able to fetch them.

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -1165,7 +1165,7 @@ if config_clean?
     mk.print(<<~EOF)
 
       all: clean-ports
-      clean-ports: $(DLLIB)
+      clean-ports: $(TARGET_SO)
       \t-$(Q)$(RUBY) $(srcdir)/extconf.rb --clean --#{static_p ? "enable" : "disable"}-static
     EOF
   end

